healthtools/scrapers/base_scraper.py

```
@@ -0,0 +1,189 @@
+            except Exception as err:
+                skipped_pages += 1
+                print "ERROR: scrape_site() - source: {} - page: {} - {}".format(url, page_num, err)
+                continue
+        print "| {} completed. | {} entries retrieved. | {} pages skipped.".format(type(self).__name__, len(all_results), skipped_pages)
+
+        if all_results:
+            all_results_json = json.dumps(all_results)
+            delete_batch = json.dumps(delete_batch)
+
+            self.delete_cloudsearch_docs()
+            self.upload_data(all_results_json)
+            self.archive_data(all_results_json)
+
+            # store delete operations for next scrape
+            delete_file = StringIO(delete_batch)
+            self.s3.upload_fileobj(
+                delete_file, "cfa-healthtools-ke",
+                self.delete_file)
+
+            return all_results
+
+    def scrape_page(self, page_url):
+        '''
+        Get entries from page
+        '''
+        try:
+            soup = self.make_soup(page_url)
+            table = soup.find('table', {"class": "zebra"}).find("tbody")
+            rows = table.find_all("tr")
+
+            entries = []
+            delete_batch = []
+            for row in rows:
+                # only the columns we want
+                # -1 because fields/columns has extra index; id
+                columns = row.find_all("td")[:len(self.fields) - 1]
+                columns = [text.text.strip() for text in columns]
+                columns.append(self.generate_id())
+
+                entry = dict(zip(self.fields, columns))
+                entry = self.format_for_cloudsearch(entry)
+                entries.append(entry)
+
+                delete_batch.append({"type": "delete", "id": entry["id"]})
+                self.document_id += 1
+            return entries, delete_batch
+        except Exception as err:
+            print "ERROR: Failed to scrape data from page {}  -- {}".format(page_url, str(err))
```

Should also return an error variable.